packages/core/admin/server/src/controllers/webhooks.ts

```diff
@@ -137,7 +137,20 @@ export default {
 
     const webhook = await strapi.webhookStore.findWebhook(id);
 
-    const response = await strapi.webhookRunner.run(webhook as Webhook, 'trigger-test', {});
+    const response = await strapi.webhookRunner
+      .run(webhook as Webhook, 'trigger-test', {})
+      .then((res: { statusCode: number; message: string }) => {
+        // Mask the response to avoid leaking sensitive information
+        if (+res.statusCode > 300) {
+          strapi.log.error(`Failed to manually trigger webhook: \n${JSON.stringify(res, null, 2)}`);
+          return {
+            statusCode: 500,
+            message: 'Webhook URL request failed',
+          };
+        }
+
+        return res;
+      });
```

Comment:
```suggestion
    let response = await strapi.webhookRunner.run(webhook as Webhook, 'trigger-test', {});

    if (Number(response.statusCode) > 300) {
      strapi.log.error(`Failed to manually trigger webhook: \n${JSON.stringify(response, null, 2)}`);
      response = {
        statusCode: 500,
        message: 'Webhook URL request failed',
      };
    }
```
(just to simplify it)

Are we sure we don't need to suppress any success codes?

Reply:
which information could be leaked by other 200 status code?
well, I guess it would also make sense to mask it to 200 just in case, will do that